about-pses/example-pse.md

```diff
@@ -1,90 +1,170 @@
 
-Imagine working on software that deals with restaurant reviews. Create a function named `get_highest_rated` that is responsible for finding the highest-rated restaurant. This function should take in a list of dictionaries called `restaurants` as a parameter. Each dictionary represents the data associated with a restaurant, including its rating. This function should have a return value of the restaurant with the highest rating.
+Create a function named `winner` that takes two arguments, the move for `player_1` and `player_2` (`"rock"`, `"paper"`, or `"scissors"`), and returns the winner (`"Player 1 wins!"`, `"Player 2 wins!"`, or `"It's a tie"`).
 
-Answer the following prompts:
+## Example Input/Output
 
-1. List five or more questions whose answers would clarify the problem statement
-2. List two sets of example arguments and the expected return value for these arguments
-3. Divide the project prompt into at least two different sub-problems
-4. Pick one of those problems from #3 and write a numbered list explaining how to solve that problem without using full lines of code
-5. Write the pseudocode or code solution for your answer for #4
+| Input (Argument of the function) | Expected Output (Return value of the function) | Explanation|
+|--|--|--|
+|player_1 = `"rock"` <br> player_2 = `"scissors"`| `"Player 1 wins!"`|`"rock"` beats `"scissors"`|
+|player_1 = `"rock"` <br> player_2 = `"rock"`| `"It's a tie!"`|`player_1` and `player_2` played the same move|
 
-## Example Answers
+## Rock, Paper, Scissors Ruleset
+|Player 1|Player 2|Winner|
+|---|---|---|
+|Rock|Rock|Tie|
+|Paper|Paper|Tie|
+|Scissors|Scissors|Tie|
+|Rock|Paper|Player 2|
+|Rock|Scissors|Player 1|
+|Paper|Rock|Player 1|
+|Paper|Scissors|Player 2|
+|Scissors|Rock|Player 2|
+|Scissors|Paper|Player 1| 
 
-### #1 Five or More Clarifying Questions
+## Prompts
 
-1.  Should I return the restaurant's name, or the dictionary that represents the restaurant data, or the rating?
-1.  What is the name of the key in each dictionary whose value is the rating?
-1.  What kind of data-type is rating stored in?
-1.  What should happen if there's more than one highest-rated restaurant?
-1.  Is the restaurant's name included in this dictionary?
-1.  What else is included in each dictionary?
-1.  Will there ever be zero restaurants? What should happen?
-1.  What should I do if there is no rating?
+### Ask Clarifying Questions
 
-### #2 Two Example Inputs/Outputs
+List five or more questions whose answers would clarify the problem statement.
 
-1.  Example one
-    - Input: `[{name: "Grillby's", rating: 1}, {name: "Crow's Nest", rating: 5}]`
-    - Output: `{name: "Crow's Nest", rating: 5}`
-1.  Example two
-    - Input: `[{name: "Crow's Nest", rating: 1}]`
-    - Output: `{name: "Crow's Nest", rating: 1}`
+```
+1. How should the function handle invalid user input (i.e. player_1 = "lizards")?
+2. 
+3.
+4.
+5. 
```

Comment:
Additional questions:
- Should a user automatically lose if they have invalid input?
- What should happen if both users have invalid input?
- Does capitalization matter?
- Does extra whitespace / punctuation matter?
- Should anything be printed to the console?
- Is the speed / memory usage of this function important?
- Should there be a default value for each of the arguments?

...but this question is so small and well-defined that coming up with this many clarifying questions _is_ really hard. Maybe for this one say that they only need to come up with one or two clarifying questions for this PSE but that later problems will be larger and they'll be expected to come up with more?